### PR TITLE
Feature/request includes

### PR DESCRIPTION
**Pull request by digitalsadhu**
@tsteuwer 
I'm creating a PR for your code to make it easier to review/discuss.

### CODE REVIEW

File: lib/deserialize.js [hunk 1]
```diff
@@ -49,6 +49,25 @@ module.exports = function (app) {
...
+			if (!RelUtils.shouldIncludeRelationships(ctx.req.method)) {
+				return next(RelUtils.getInvalidIncludesError());
+			}
+		
+			var include = RelUtils.getIncludesArray(ctx.req.query);
+			include = include.length === 1 ? include[0] : include;
+
+			ctx.req.remotingContext.args = ctx.req.remotingContext.args || {};
+			ctx.req.remotingContext.args.filter = ctx.req.remotingContext.args.filter || {};
+			ctx.req.remotingContext.args.filter.include = include;
```
Comment (digitalsadhu) [inline lib/deserialize.js]: I have a sneaking suspicion that ctx.req.remotingContext is a circular reference to ctx. Will follow up after looking through the rest of the code

**Comment by digitalsadhu** [comment]
@tsteuwer nice progress mate!
I've cleaned up the eslint errors and such. Getting a jsinspect copy/paste code duplication type error. Might be worth some refactoring to clean that up.

File: test/hasManyRelationships.test.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            title: 'my post',
+            content: 'my post content',
+            author: 1,
+            comments: [1, 2]
+          });
+          expect(res.body.included).to.be.an('array');
+          expect(res.body.included.length).to.equal(3);
+          expect(res.body.included[0]).to.deep.equal({
+            id: '1',
+            type: 'author',
```
Comment (digitalsadhu) [inline test/hasManyRelationships.test.js]: type should always be plural, so in this case 'authors'

**Comment by tsteuwer** [comment]
Is that part of the json api spec? I don't remember reading that. I thought it only uses plurals in the fetch many requests?

**Comment by digitalsadhu** [comment]
No, in our case type should be plural.

The spec says:
> Note: This spec is agnostic about inflection rules, so the value of type can be either plural or singular. However, the same value should be used consistently throughout an implementation.

Also, have a look through the naming recommendations http://jsonapi.org/recommendations/ 
note, collection urls should be named from the type: /photos and /photos/1 or /photo and /photo/1

So:
- Key thing is consistency, we use plural as type throughout the rest of the component so we need to stick with that.
- All examples on jsonapi for both fetching a single resource and fetching a collection use plural
- loopback urls use plural by default so good to have type match that.
- The type is always the same and shouldn't change between single resource fetch and collection fetch.

**Comment by tsteuwer** [comment]
Sounds good to me. I'll make the changes :)

**Comment by digitalsadhu** [comment]
Saaweeet. I had a good play with it last night. Looking good! Will be an
awesome to get done. I feel like we might be able to drop the alpha
recommendation and go with beta after we have this feature. What do you
think?

**Comment by tsteuwer** [comment]
@digitalsadhu, plural has been fixed and merged.  I have no idea what travis is screaming about though? It gives no error, just says "1 match" but no actual details. Any ideas? Also, yeah, lets stick an Alpha tag on this bad-boy!!!

**Comment by digitalsadhu** [comment]
nice @tsteuwer !
Yea I was confused by the failing at first as well. It turns out there is an extra check defined in .travis.yml 
The check is a copy and paste check via jsinspect here https://github.com/digitalsadhu/loopback-component-jsonapi/blob/master/.travis.yml#L16
We could just disable this check but at the same time we could improve the code by pulling out the code into a function and calling it in a loop to avoid the code duplication.

**Comment by digitalsadhu** [comment]
@tsteuwer What do you think about essentially having the function:
`handleSingleInclude` which defines the main logic and then `handleMultipleIncludes` which loops through calling `handleSingleInclude` for each item?

**Comment by taras** [comment]
@digitalsadhu I'll give a shot doing the refactor that you're suggesting

**Comment by taras** [comment]
@tsteuwer what else do we need to ship this?

**Comment by digitalsadhu** [comment]
@taras
The checklist is here
https://github.com/digitalsadhu/loopback-component-jsonapi/issues/38

Slightly out of date now as most of the items can probably be checked off except the 400 error. It's possible also that loopbacks original syntax for include stops working in favor of json API compliant syntax. I think we should probably just support both.

**Comment by tsteuwer** [comment]
I am pretty sure this just works with the current filter option. I parse the json api include spec into the loopback version then just format it for json api

**Comment by tsteuwer** [comment]
@taras @digitalsadhu I think all I need to so now is rebase from master to get the latest and we should be good to merge unless one of you can. I'm in Colorado right now visiting family. Sorry for the long wait. Holidays have me traveling a lot to see family

**Comment by tsteuwer** [comment]
@taras @digitalsadhu We are officially go-to-go! All checks have passed and is up-to-date with ``master``. @digitalsadhu, if you would do the honors of merging and creating a new release? Great job everyone. This is probably one of the most requested features I can think of.

**Comment by digitalsadhu** [comment]
Ok, @taras @tsteuwer I've done a bit more testing and fixing. 
The checklist here: https://github.com/digitalsadhu/loopback-component-jsonapi/issues/38
is now complete.

**Comment by tsteuwer** [comment]
@digitalsadhu great job. We're ready to merge and create a release? Once done I can go and update the emberloop repo and get the tests written

**Comment by digitalsadhu** [comment]
@tsteuwer 
Just doing a little manual testing. I'm seeing an issue with using include with a related url:
eg.
GET /api/posts/9/author?include=posts
produces an error.

It's probably worth logging an issue and proceeding rather than holding things up any longer

**Comment by tsteuwer** [comment]
@digitalsadhu. Yeah go and make a new ticket and I'll work on it once the release is out and I write the tests for emberjs

**Comment by digitalsadhu** [comment]
:+1: 
Alright, lets do this.

**Comment by tsteuwer** [comment]
@digitalsadhu also make sure you publish to npm so I can update via npm :)

**Comment by digitalsadhu** [comment]
@tsteuwer @taras 
Merged and published as v0.11.0!
Well done guys!

**Pull request merged**